src/controllers/userProfileController.js

```diff
@@ -269,8 +269,9 @@ const userProfileController = function (UserProfile) {
         || req.body.requestor.requestorId === userid
       )
     );
+    const canUserToggleInvisibility = await hasPermission(req.body.requestor, 'toggleInvisibility');
+    if (!isRequestorAuthorized && !canUserToggleInvisibility) {
```

Comment:
Pseudocode
```suggestion
    if (canUserToggleInvisibility && user.invisible !== request.user.invisible) {
        update user.invisible
    }
    if(!isRequestorAuthorized) {
```
Need to separate so `canUserToggleInvisibility` does not give the same access as `canPutUserProfile`